packages/nimble-components/src/wafer-map/types.ts

```diff
@@ -30,6 +30,7 @@ export interface WaferMapDie {
     value: string;
     x: number;
     y: number;
+    tooltip: string;
```

Comment:
Is it the case that dies must always have a tooltip? If not we could make it optional:
https://www.typescriptlang.org/docs/handbook/interfaces.html#optional-properties

Then the code that uses it would have to check whether it was present but clients wouldn't have to provide it. This seems preferable to me, unless the tooltip is truly required. It would also reduce the number of files that need to change in this PR.

Reply:
tooltips were explicitly cited as a non-goal for the wafer map (to be handled in the plugin instead): https://github.com/ni/nimble/tree/main/packages/nimble-components/src/wafer-map/specs#non-goals

So we haven't discussed api considerations yet.

My biggest questions:
1. How will tooltips be rendered, in the canvas itself or as html positioned over the canvas
2. For canvas based tooltips how will we ensure visual consistency with nimble's html based tooltips? how will accessibility of tooltip strings work?
3. For html based tooltip will there be a need for rich tooltips beyond simple strings? For rich tooltips we likely would necessitate an api choice like the [table is doing for action menus](https://github.com/ni/nimble/pull/956) which solves a similar set of problems (slotting arbitrary content on an action, in this case a hover).

I think we should potentially have a spec update to make sure we are clear on the api and change in scope of what will be in nimble.

Reply:
Applied the change.
To answer your questions - the tooltip display will be handled within the plugin. As it is, we want to be able to store tooltip information (if available) for each individual die, within the component. During a `hover ` event over a given die, we want to be able to retrieve the corresponding tooltip information and pass it to the containing plugin, which in turn will handle the actual display/render operation - the Mictrostrategy library does provide some API in this sense.

Reply:
Thanks for the context @DStavilaNI. If this field won't ever be used by the wafer map itself, I wonder if it should be made more generic. We could give it a more generic name like `metadata` and a type of `unknown` or a generic type so that client applications can provide whatever type of metadata they want. The microstrategy plugin would start by only providing a tooltip string but eventually it could store other info like a link to navigate to when clicked. This would avoid having to change Nimble every time a client wanted to store different information.

Thoughts?

Reply:
Good point ... we will apply this.